src/spikeinterface/widgets/crosscorrelograms.py

```diff
@@ -46,7 +46,9 @@ def __init__(
         backend=None,
         **backend_kwargs,
     ):
-        sorting_analyzer_or_sorting = self.ensure_sorting_analyzer(sorting_analyzer_or_sorting)
+
+        if not isinstance(sorting_analyzer_or_sorting, BaseSorting):
+            sorting_analyzer_or_sorting = self.ensure_sorting_analyzer(sorting_analyzer_or_sorting)
```

Comment:
Here you break the fact that sorting can be an input no ?

Reply:
Actually tests are passing now. 

The logic for the if statement is 

if base sorting => continue as base sorting
if not base sorting => ensure sorting analyzer (for sorting analyzer or mock waveform)

So both cases are addressed.

Reply:
ok
then it is ok.